src/sif/dynamiccost.cc

```diff
@@ -168,6 +168,13 @@ DynamicCost::DynamicCost(const Costing& costing,
   for (auto& edge : costing.options().exclude_edges()) {
     user_exclude_edges_.insert({GraphId(edge.id()), edge.percent_along()});
   }
+
+  // set all max_up_transitions to kUnlimitedTransitions if disable_hierarchy_pruning is true
+  if (costing.options().disable_hierarchy_pruning()) {
+    for (auto& h : hierarchy_limits_) {
+      h.max_up_transitions = kUnlimitedTransitions;
+    }
+  }
```

Comment:
you can consolidate this with the hierarchy limit stuff a few lines up (line 163 currently)

Reply:
done!